### PR TITLE
Add more decode tests

### PR DESCRIPTION
**Pull request by tirr-c**
Adds tests for regular images and regression tests. Tests if samples match in 16-bit integer (error of ~1.5e-5).

### CODE REVIEW

**Comment by EugeneVert** [comment]
Could these tests be added as a git submodule to prevent the repository from becoming too large?

File: crates/jxl-oxide/tests/decode.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                    Ok(expected) => {
+                        let expected = zstd::Decoder::new(expected).expect("Failed to open Zstandard stream");
+                        decode(&input, expected);
+                    },
+                    Err(_) => {
+                        let mut output_url = path.clone();
+                        output_url.push("output.buf.zst.url");
+                        let output_url = std::fs::read_to_string(output_url).expect("Failed to get fixture URL");
+
+                        let resp = reqwest::blocking::get(output_url.trim()).expect("Failed to connect");
```
Comment (EugeneVert) [inline crates/jxl-oxide/tests/decode.rs:99]: We would probably want to do caching here, like in `conformance.rs`

**Comment by EugeneVert** [comment]
Also, `issue_26` test fails on **debug** build. I guess it's okay, but just as a note
```
thread 'issue_26' panicked at 'assertion failed: `(left == right)`
  left: `62152`,
 right: `62153`', crates/jxl-oxide/tests/decode.rs:55:25
```

File: crates/jxl-oxide/tests/decode.rs (hunk 1 above)
Comment (tirr-c) [reply crates/jxl-oxide/tests/decode.rs]: Good point! Added caching for downloaded image buffers.

**Review by EugeneVert — APPROVED** [review]

**Comment by EugeneVert** [comment]
Fixed https://github.com/tirr-c/jxl-oxide/issues/28?

**Comment by tirr-c** [comment]
> Fixed #28?

Oh, missed that!

**Pull request merged**